File: backend/OnboardingDiary.Application/Issues/Dtos/IssueDto.cs
namespace OnboardingDiary.Application.Issues.Dtos;

public record IssueDto(
    Guid Id,
    Guid UserId,
    DateTime Date,
    string Title,
    string Description,
    string Severity,
    string Status,
    string? ResolutionNotes,
    DateTime? ResolvedAt,
    bool IsEscalated,
    DateTime CreatedAt,
    DateTime UpdatedAt);

File: src/main/java/com/codev/onboardingdiary/web/dto/IssueDto.java
package com.codev.onboardingdiary.web.dto;

import com.codev.onboardingdiary.domain.IssueStatus;
import com.codev.onboardingdiary.domain.Severity;
import java.time.LocalDate;

public record IssueDto(Long id,
                       LocalDate date,
                       String title,
                       String description,
                       Severity severity,
                       IssueStatus status,
                       String resolutionNotes) {
}
